File: 第3阶段-运维线上实战：千万级电商系统解决方案/12_企业架构之Tomcat部署使用/04_代码/脚本检测主从/slave.py
#!/bin/python
import MySQLdb
db = MySQLdb.connect(host='127.0.0.1',user='root',passwd='123456')
cur = db.cursor()
cur.execute('show slave status')
#print(cur)
for row in cur.fetchall():
 #print(row)
 #print(row[10])
 #print(row[11])
 if row[10]=='Yes' and row[11]=='Yes':
    print('slave is ok') 
db.close
#dir(pymysql)

File: 第3阶段-运维线上实战：千万级电商系统解决方案/12_企业架构之Tomcat部署使用/04_代码/脚本检测主从/slave.php
<?php
$mysqli = new mysqli('127.0.0.1','root','123456');
$sql = "show slave status";
$rs = $mysqli->query($sql);
#var_dump($rs);
$data = $rs->fetch_assoc();
#var_dump($data);
if ($data['Slave_IO_Running']=='Yes' && $data['Slave_SQL_Running'] == 'Yes'){
  echo 'slave is ok'."\n";
}else{
  echo 'Last_IO_Error:'.$data['Last_IO_Error']."\n"; 
}
